draft-ietf-taps-interface.md

```diff
@@ -1245,36 +1250,32 @@ Connection.Batch(
 )
 ~~~
 
-## Send on Active Open: InitiateWithIdempotentSend {#initiate-and-send}
+## Send on Active Open: InitiateWithSend {#initiate-and-send}
 
 For application-layer protocols where the Connection initiator also sends the
-first message, the InitiateWithIdempotentSend() action combines Connection initiation with
-a first Message sent, provided that message is idempotent.
+first message, the InitiateWithSend() action combines Connection initiation with
+a first Message sent.
 
 Without a message context (as in {{send-basic}}):
 
 ~~~
-Connection := Preconnection.InitiateWithIdempotentSend(messageData)
+Connection := Preconnection.InitiateWithSend(messageData)
 ~~~
 
 With a message context (as in {{message-props}}):
 
 ~~~
-Connection := Preconnection.InitiateWithIdempotentSend(messageData, messageContext)
+Connection := Preconnection.InitiateWithSend(messageData, messageContext)
 ~~~
 
-The message passed to InitiateWithIdempotentSend() is, as suggested by the
-name, considered to be idempotent (see {{msg-idempotent}}) regardless of
-declared message properties or defaults. If protocol stacks supporting 0-RTT
-establishment with idempotent data are available on the Preconnection, then
-0-RTT establishment may be used with the given message when establishing
-candidate connections. For a non-idemponent initial message, or when the
-selected stack(s) do not support 0-RTT establishment,
-InitiateWithIdempotentSend is identical to Initiate() followed by Send().
+Whenever possible, a messageContext should be provided to declare the message passed to InitiateWithSend
+as idempotent. This allows the transport system to make use of 0-RTT establishment in case this is supported
+by the available protocol stacks. When the selected stack(s) do not support transmitting data upon connection
+establishment, InitiateWithSend is identical to Initiate() followed by Send().
```

Comment:
Perhaps just a question to understand this:
What if the InitiateWithSend() message is not marked as idempotent, but the protocol stack supports 0-RTT?
In this case, no 0-RTT happens, right?

Reply:
Exactly